api_docs/changelog.md

```diff
@@ -20,6 +20,20 @@ format used by the Zulip server that they are interacting with.
 
 ## Changes in Zulip 10.0
 
+**Feature level 354**
+* [`GET /messages`](/api/get-messages),
+  [`GET /messages/{message_id}`](/api/get-message),
+  [`POST /messages/flags/narrow`]: Users can access messages of private
+  channels without subscribing if they are part of groups that grant
+  content access.
+* [`POST /messages/{message_id}/reactions`](/api/add-reaction),
+  [`DELETE /messages/{message_id}/reactions`](/api/remove-reaction):
+  Users can react to messages of private channels without subscribing
+  if they are part of groups that grant content access.
+* [`PATCH /messages/{message_id}`](/api/update-message): Users can edit
+  messages they have posted in a private channel after unsubscribing if
+  they are part of groups that grant content access.
```

Comment:
This list is I think not complete; read receipts, submessages (responding to polls), typing notifications for editing messages all go through this code path.

Reply:
https://github.com/zulip/zulip/pull/33507#discussion_r1966280338